### ISSUE TITLE
strict dereferencing result of proc call or list index fails

### ISSUE THREAD

**Issue by willox** [wixoaGit/OpenDream#114]
```dm
/datum/test
	var/val = 0

	proc/self()
		return src

/proc/do_test()
	var/datum/test/T1 = new()
	return T1.self().val
```

Error at code.dm:9:22: Expected dedent
Error at code.dm:9:22: Expected EOF

**Comment by ike709** [comment]
BeeStation blocker

**Comment by wixoaGit** [comment]
As of 82cebc03172f1dad83513dbea01d96151c915690 this can be parsed, but if it's a strict dereference (`.` instead of `:`) it errors due a lack of type.

**Comment by wixoaGit** [comment]
82cebc03172f1dad83513dbea01d96151c915690 was reverted. aad010043572f94fcc0370a51da8dd20401e2e53 is the second attempt at this, with (hopefully) no issues with ternary expressions. The previous problem with strict dereferencing still exists.

**Comment by wixoaGit** [comment]
Fixed by #440
